### DISCUSSION TITLE
Redact error message for client for status 500 errors

### DISCUSSION THREAD

**Discussion by lgraubner** [trpc/trpc#2071] [General]
I noticed the client side error message for unhandled server side errors contains the original error message. I don't think that is a good idea, at least for my case as those can be quite sensitive information. Is there any way to redact/overwrite it?

The [error handling docs](https://trpc.io/docs/error-handling#handling-errors) mention it's possible to alter errors, but I couldn't figure out how to do it. I guess that's the right spot to do something like this?

**Comment by sachinraja** [comment]
It's only shown when `process.env.NODE_ENV !== 'production'`. Feel free to send a PR to clarify this in the docs

**Comment by lgraubner** [comment]
Thanks for the prompt response. I thought I saw the message in production, that's why I'm asking. But I might be mistaken, will have a look for it and reopen if it's still the case.

**Comment by lgraubner** [comment]
I just tested this manually and my initial observation seems to be true:

```
router.query('createBooking', {
    async resolve() {
        throw new Error('this should not be visible');
    }
})
```

Client shows this:
<img width="438" alt="Bildschirmfoto 2022-07-01 um 18 04 33" src="https://user-images.githubusercontent.com/2519959/176930901-6a13d6c8-442a-4e3d-b0ef-3f68c4d9bf6b.png">

It's definitely running with `NODE_ENV=production`.

**Comment by KATT** [comment]
You can do custom error formatting if you want to change the behavior. I think propagating the error message is a sane default?

**Reply by sachinraja** [reply]
Here's an example @lgraubner:
```ts
export const appRouter = trpc
  .router()
  .formatError(({ error, shape }) => {
    if (error.code === 'INTERNAL_SERVER_ERROR') {
      return {
        ...shape,
        message: 'Internal server error',
      }
    }
    return shape
  })
```

**Reply by lgraubner** [reply]
I'd opt for masking the error as default. As those errors are probably always unintentionally it's quite easy to leak internal stuff.

Formatting the error will do for me for now, thanks!
